spec/index.html

```diff
@@ -0,0 +1,553 @@
+        <li>User agent 关闭执行引擎 Logic Layer</li>
+      </ol>
+    
+      <ol its-locale-filter-list="en" lang="en">
+        <li> Trigger MiniApp widget running in background event</p>
+        <li> The user agent closes the window where the widget is located and the rendering engine View Layer</p>
+        <li> Trigger the MiniApp widget unloading event</p>
+        <li> The user agent closes the execution engine Logic Layer</p>
+      </ol>
+    
+    </section>
+
+    <section id='foreground-background-process'>
+      <h3>
+      <span its-locale-filter-list="en" lang="en">MiniApp widget switch between foreground and background process</span>
+      <span its-locale-filter-list="zh-hans" lang="zh-hans">Widget在前台和后台见切换过程</span>  
+      </h3>
+    
+      <p its-locale-filter-list="zh-hans" lang="zh-hans">MiniApp widget 相比于 MiniApp page ，会因为更多的原因不可见，这时 MiniApp widget 的
+        running in background 事件会被触发，例如：</p>
+      <p its-locale-filter-list="en" lang="en">Compared with MiniApp page, MiniApp widget will be invisible for more
+        reasons. The the running in background event of MiniApp widget will be triggered, for example: </p>
+    
+      <ol its-locale-filter-list="zh-hans" lang="zh-hans">
+        <li> widget 被用户移出可见区域</li>
+        <li> widget 被折叠</p>
+        <li> widget 被其他 Widget 完全覆盖</li>
+        <li> widget 所在的 [=host environment=] 进入 background </li>
+      </ol>
+    
+      <ol its-locale-filter-list="en" lang="en">
+        <li>The widget is moved out of the visible area by user </li>    
+        <li>Widget is collapsed </li>    
+        <li>Widget is completely covered by other widget </li>    
+        <li>The [=host environment=] where the widget is located enters the background </li>
+      </ol>
+        
+      <p its-locale-filter-list="zh-hans" lang="zh-hans">当 MiniApp widget 重新可见时，running in foreground 事件会被触发。</p>    
+      <p its-locale-filter-list="en" lang="en">The running in foreground event would be triggered when MiniApp widget becomes visible again.</span></p>
+    </section>
+
+    <section id='update-process'>
+      <h3>
+      <span its-locale-filter-list="en" lang="en">MiniApp widget update process</span>
+      <span its-locale-filter-list="zh-hans" lang="zh-hans">Widget更新过程</span>   
+      </h3>
+
+      <p its-locale-filter-list="zh-hans" lang="zh-hans">User agent 可以在不重新创建 Widget 环境的情况下更新 widget 的内容，这种情况常应用于 Widget 的 Host environment 与 Widget 有强交互的情况，
+        例如，在商品搜索结果页显示一个商品 widget 时，用户点击另一个商品，该 Widget 的内容将被更新为新商品的。</p>
+      <p its-locale-filter-list="en" lang="en">User agent can update the content of the
+        widget without recreating the widget environment. This often happens where the
+        widget host environment has strong interaction with the widget. For example,
+        when a product widget is displayed on search resualts page, user clicks on
+        another product, the content of the widget will be updated with the infomation of
+        the new product.</p>
+
```

Comment:
It seems not clear what the strong interaction means here. In the above example "when a product widget is displayed on search results page, user clicks on another product, the content of the widget will be updated with the infomation of the new product", it is more like clicking a product link and then opening the new link, which seems quite usual in Web world. What is the motivation to reuse the existing wiget environment, instead of creating a new one?

Also, suggest to include how to update the widget. Request and fetech new xml, css, js and image resources from server, and replace the existing resources?

And, if user wants to go back to the results page, how to switch back?